src/Moq/ProxyFactories/CastleProxyFactory.cs

```diff
@@ -101,6 +101,15 @@ public void Intercept(Castle.DynamicProxy.IInvocation underlying)
 					this.interceptor.Intercept(invocation);
 					underlying.ReturnValue = invocation.ReturnValue;
 				}
+				catch (MockException)
+				{
+					throw;
+				}
+				catch (Exception ex)
+				{
+					invocation.ReturnValue = new InvocationExceptionWrapper(ex);
+					throw;
```

Comment:
I think `throw`ing the original exception is the right thing to do here:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/throw#re-throwing-an-exception